plugins/cache/redis.py

```diff
@@ -91,6 +92,8 @@ class CacheModule(BaseCacheModule):
     performance.
     """
     _sentinel_service_name = None
+    re_url_conn = re.compile(r'^(.*):(\d+):(\d+)(?::(.*))?$')
```

Comment:
This regex can split `1:2:3:4` as `'1:2'`, `'3'`, `'4'`, `''`. This is not what you intend.

How about using the following instead:
```suggestion
    re_url_conn = re.compile(r'^([^:]+|\[[^]]+\]):(\d+):(\d+)(?::(.*))?$')
```